cmd/osquery-perf/agent.go

```diff
@@ -274,20 +291,168 @@ func (a *agent) runLoop(i int, onlyAlreadyEnrolled bool) {
 	}
 }
 
+func (a *agent) runOrbitLoop() {
+	orbitClient, err := service.NewOrbitClient(
+		"",
+		a.serverAddress,
+		"",
+		false,
+		a.EnrollSecret,
+		a.UUID,
+	)
+	if err != nil {
+		log.Println("creating orbit client: ", err)
+	}
+
+	orbitClient.TestNodeKey = *a.orbitNodeKey
+
+	deviceClient, err := service.NewDeviceClient(a.serverAddress, false, "")
+	if err != nil {
+		log.Println("creating device client: ", err)
+	}
+
+	// orbit does a config check when it starts
+	if _, err := orbitClient.GetConfig(); err != nil {
+		log.Println("orbitClient.GetConfig: ", err)
```

Comment:
Should we record stats when there are failures in all these endpoints?
```
a.stats.RecordStats(...)
```
Maybe two new stats `orbitErrors` and `desktopErrors`.

Reply:
that sounds good to me, I will add both.